src/main/java/org/traccar/protocol/NavtelecomProtocolDecoder.java

```diff
@@ -221,6 +228,159 @@ protected Object decode(
+                                        position.set("rs485fuel_level" + 1,tmp);
+                                    }
+                                    break;
+                                case 39:
+                                    tmp = buf.readUnsignedShortLE();
+                                    if (tmp < 65500) {                                                                      // Do not write error codes
+                                        position.set("rs485fuel_level" + 2,tmp);
+                                    }
+                                    break;
+                                case 40:
+                                    tmp = buf.readUnsignedShortLE();
+                                    if (tmp < 65500) {                                                                      // Do not write error codes
+                                        position.set("rs485fuel_level" + 3,tmp);
+                                    }
+                                    break;
+                                case 41:
+                                    tmp = buf.readUnsignedShortLE();
+                                    if (tmp < 65500) {                                                                      // Do not write error codes
+                                        position.set("rs485fuel_level" + 4,tmp);
+                                    }
+                                    break;
+                                case 42:
+                                    tmp = buf.readUnsignedShortLE();
+                                    if (tmp < 65500) {                                                                      // Do not write error codes
+                                        position.set("rs485fuel_level" + 5,tmp);
+                                    }
+                                    break;
+                                case 43:
+                                    tmp = buf.readUnsignedShortLE();
+                                    if (tmp < 65500) {                                                                      // Do not write error codes
+                                        position.set("rs485fuel_level" + 6,tmp);
+                                    }
+                                    break;
+                                case 44:
+                                    tmp = buf.readUnsignedShortLE();
+                                    if (tmp < 65500) {                                                                      // Do not write error codes
+                                        position.set("rs232fuel_level",tmp);
+                                    }
+                                    break;
+                                case 45:
+                                    tmp = buf.readByte();
+                                    if (tmp != 0x80) {                                                                      // Do not write error codes
```

Comment:
Same here. Extract checks into a separate function. Also remove the comments.

Reply:
See, maybe this is a good option.